src/ui/component/filePicker/filePicker.tsx

```diff
@@ -22,6 +21,38 @@ type FilePickerProps = {
   onError?: (error: Error) => void
 }
 
+const extractFileStream = (file: globalThis.File): File =>
+  Object.assign(file, {
+    stream: file.stream()
+  })
+
+const mapDataTransferItemToFiles = (item: DataTransferItem): File[] => {
+  const file = item.getAsFile()
+  if (!file) throw new Error('Impossible to get file.')
+  return [extractFileStream(file)]
+}
+
+const traverseDirectory = async (fsDirectory: FileSystemDirectoryEntry): Promise<File[]> => {
+  const files: File[] = []
+  const directoryReader = fsDirectory.createReader()
+
+  const entries = await new Promise<FileSystemEntry[]>(resolve =>
+    directoryReader.readEntries(resolve)
+  )
```

Comment:
The `readEntries` method of the File System API can return directory entries in chunks, especially when dealing with directories that contain a large number of files or subdirectories. This is to ensure efficient memory usage and maintain application performance.

We can implement a loop or recursive mechanism to keep calling readEntries until it returns an empty array. This will ensure that you have fetched all entries for a given directory.

You can find more explanations here:
https://stackoverflow.com/questions/51850469/file-and-directory-entries-api-broken-in-chrome

Reply:
Indeed, thank you for this constructive feedback.  🙏
I modified the code accordingly and it works well, even by dropping a folder of `node_modules` ^^ 
It takes time, that's why it will be necessary to add a loader (to do with the product team). 
Also, we see the limits of storing the contents of the file, it would be necessary to make the changes in a second time as you had seen together with @fredericvilcot.

Reply:
Fantastic!